src/dialogs/ShuffleDialog.java

```diff
@@ -0,0 +1,394 @@
+
+        this.jButton1 = new javax.swing.JButton();
+        this.jButton2 = new javax.swing.JButton();
+
+        this.jScrollPane4 = new javax.swing.JScrollPane();
+        this.jList4 = new javax.swing.JList();
+        this.jLabel4 = new javax.swing.JLabel();
+
+        this.jScrollPane1 = new javax.swing.JScrollPane();
+        this.jList1 = new javax.swing.JList();
+        this.jLabel1 = new javax.swing.JLabel();
+
+        this.jScrollPane3 = new javax.swing.JScrollPane();
+        this.jList3 = new javax.swing.JList();
+        this.jLabel3 = new javax.swing.JLabel();
+
+        this.jScrollPane2 = new javax.swing.JScrollPane();
+        this.jList2 = new javax.swing.JList();
+        this.jLabel2 = new javax.swing.JLabel();
+
+        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
+
+        jButton1.setText("Import...");
+        jButton1.addActionListener(new java.awt.event.ActionListener() {
+            @Override
+            public void actionPerformed(java.awt.event.ActionEvent evt) {
+                jButton1ActionPerformed();
+            }
+        });
+        
+        jButton2.setText("Export...");
+        jButton2.addActionListener(new java.awt.event.ActionListener() {
+            @Override
+            public void actionPerformed(java.awt.event.ActionEvent evt) {
+                jButton2ActionPerformed();
+            }
+        });
+
+        jScrollPane4.setHorizontalScrollBarPolicy(javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
+        jScrollPane4.setViewportView(this.jList4);
+
+        jScrollPane1.setHorizontalScrollBarPolicy(javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
+        jScrollPane1.setViewportView(this.jList1);
+
+        jScrollPane3.setHorizontalScrollBarPolicy(javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
+        jScrollPane3.setViewportView(this.jList3);
+        
+        jScrollPane2.setViewportView(this.jList2);
+        jScrollPane2.setHorizontalScrollBarPolicy(javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
+
+        jList4.addListSelectionListener(new javax.swing.event.ListSelectionListener() {
+            @Override
+            public void valueChanged(javax.swing.event.ListSelectionEvent evt) {
+                try {
+                    jList4ValueChanged(evt);
+                } catch (Exception e) {
+                    JErrorPane.invokeErrorMessage(e);
+                }
+            }
+        });
+
+        jLabel4.setText("Base Distribution");
+
+        jList1.addListSelectionListener(new javax.swing.event.ListSelectionListener() {
+            @Override
+            public void valueChanged(javax.swing.event.ListSelectionEvent evt) {
+                try {
+                    jList1ValueChanged(evt);
+                } catch (Exception e) {
+                    JErrorPane.invokeErrorMessage(e);
+                }
+            }
+        });
+
+        jLabel1.setText("Distribution Change");
```

Comment:
This isn't vertically warping the sizes of elements. That's the other one.

Reply:
did some testing and it seems this is the one (where each element is mapped individually)

Reply:
Wdym?

Reply:
this is the one that vertically warps the sizes of elements (as opposed to horizontally, like how shuffles work)

Reply:
yeah, that's what I mean

Reply:
same applies to warp

Reply:
Not really

Reply:
with change, it will be the same if the array is linear and hasn't been changed at all (that is to say, this just points out that warping and shuffling are the same)

Reply:
With "Change", you can apply *any* shuffles to the array beforehand, and it will simply change the distribution.

Reply:
with warp, you can apply any changes to the array beforehand and it will simply change the shuffle

Reply:
Huh? That's what "Change" does, unless I'm completely not understanding what you're saying.

Reply:
what I'm saying is that any statement you make about warp can be flipped to make a statement about change

Reply:
And I still don't get it

Reply:
then I guess other point is just that "change" is too vague (for example when I thought it meant that it entirely discarded the old array)

what's wrong with "scale" anyway?

Reply:
I guess it just doesn't really *describe* it imo (at least not without thinking for 60 seconds about what it means)

Reply:
but imo "change" does so even less

Reply:
But I'm saying that we should come up with something that's not "scale" *or* "change".

Reply:
"stretch"?

Reply:
That sounds better imo

Reply:
if you think it's better

Reply:
That's not a very helpful answer

Reply:
I mean that I'd prefer it, but you can do whatever

Reply:
Should I make it "Distribution Stretch" or "Stretch Distribution"?

Reply:
the former, like how it's "Distribution Warp"